mabwiser/mab.py

```diff
@@ -569,6 +571,48 @@ def _validate(self):
                 check_true(np.isclose(sum(self.no_nhood_prob_of_arm), 1.0),
                            ValueError("no_nhood_prob_of_arm should sum up to 1.0"))
 
+    class TreeBandit(NamedTuple):
+        """Tree Bandit Neighborhood Policy.
+
+        This policy fits a decision tree for each arm using context history.
+        It uses the leaves of these trees to partition the context space into regions
+        and keeps a list of rewards for each leaf.
+        To predict, it receives a context vector and goes to the corresponding
+        leaf at each arm's tree and applies the given MAB learning policy to select a reward
+        from the rewards list at those leaves.
+        It then chooses the arm with the highest expected reward.
+
+        Attributes
+        ----------
+        tree_parameters: Dict, **kwarg
+            Parameters of the decision tree.
+            The keys must match the parameters of sklearn.tree.DecisionTreeClassifier.
+            When a parameter is not given, the default of that parameter in sklearn.tree.DecisionTreeClassifier
+            will be chosen.
+            Default value is an empty dictionary.
+
+        Example
+        -------
+            >>> from mabwiser.mab import MAB, LearningPolicy, NeighborhoodPolicy
```

Comment:
I assume you confirmed that this example works as-is and selects Arm2?

Reply:
I'll actually change its learning policy now to greedy epsilon=0 instead for a better test, but yes it works!